ArrayOperations/Arrays.cs

```diff
@@ -0,0 +1,324 @@
+            output[indexOutput] = right[indexRight];
+            indexRight++;
+            indexOutput++;
+          }
+        }
+        else if (indexLeft < left.Length)
+        {
+          output[indexOutput] = left[indexLeft];
+          indexLeft++;
+          indexOutput++;
+        }
+        else if (indexRight < right.Length)
+        {
+          output[indexOutput] = right[indexRight];
+          indexRight++;
+          indexOutput++;
+        }  
+      }
+      return output;
+    }
+ 
+    /// <summary>
+    /// Inserts a new element in an array at the specified position.
+    /// </summary>
+    /// <param name="input">Destination array.</param>
+    /// <param name="number">Number to insert in the array.</param>
+    /// <param name="index">Position at which to insert the new number.</param>
+    /// <returns>A new array with it's size increased by one and a new element located at the specified index.</returns>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The destination array has no elements.</exception>
+    /// <exception cref="ArgumentOutOfRangeException">Destination index is outside the bounds of the array.</exception>
+    public static int[] InsertAt(int[] input, int number, int index)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+
+      if (index < 0 || index > input.Length)
+      {
+        throw new ArgumentOutOfRangeException(nameof(index));
+      }
+
+      var output = new int[input.Length + 1];
+      output[index] = number;
+
+      for (var i = 0; i < input.Length; i++)
+      {
+        if (i < index)
+        {
+          output[i] = input[i];
+        }
+        else
+        {
+          output[i + 1] = input[i];
+        }
+      }
+
+      return output;
+    }
+    
+    /// <summary>
+    /// Inserts a new element at the beginning of the array.
+    /// </summary>
+    /// <param name="input">Destination array.</param>
+    /// <param name="number">The number to be inserted.</param>
+    /// <returns>A new array with it's size increased by one and a new element located at the front of the array.</returns>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The destination array has no elements.</exception>
+    public static int[] InsertFirst(int[] input, int number)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+      
+      const int index = 0;
+      var output = InsertAt(input, number, index);
```

Comment:
Don't create variables just to return them.  Do `return InsertAt(input, number, index);` instead.